File: PyWrapper.py
import os
import subprocess
from colorama import Fore, Style, init

init(autoreset=True)

# Clear the screen
def clear_screen():
    os.system('cls' if os.name == 'nt' else 'clear')

# Define the banner
def print_banner():
    print(Fore.CYAN + "*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*")
    print(Fore.CYAN + "|" + Fore.WHITE + "            __________          __________               " + Fore.CYAN + "|")
    print(Fore.CYAN + "|" + Fore.WHITE + "            \\______   \\ __  _  _\\_   ____/               " + Fore.CYAN + "|")
    print(Fore.CYAN + ":" + Fore.WHITE + "             |    |  _//  \\/ \\/  /|  __)_                " + Fore.CYAN + ":")
    print(Fore.CYAN + "." + Fore.WHITE + "             |    |   \\\\        //       \\               " + Fore.CYAN + ".")
    print(Fore.CYAN + ":" + Fore.WHITE + "  (\\_/)      |______  / \\__/\\__//______  /               " + Fore.CYAN + ":")
    print(Fore.CYAN + "|" + Fore.WHITE + " ( ~_~)             \\/" + Fore.WHITE + "PyEnv-Win Wrapper" + Fore.WHITE + "\\/1.0.0           " + Fore.CYAN + "|")
    print(Fore.CYAN + "|" + Fore.WHITE + " (>  >)                                                  " + Fore.CYAN + "|")
    print(Fore.CYAN + "*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*")

# Get the script name and arguments
script_name = os.path.basename(sys.argv[0])
file_to_execute = " ".join(os.sys.argv[1:])

# Get the Windows drive (e.g., C:)
windows_drive = os.environ.get('SystemDrive', 'C:')

# Get the username
username = os.environ.get('UserName', '')

# Installed?
pyenv = os.path.join(windows_drive + '\\Users\\' + username, '.pyenv\\')

# Check if this is the first run
if os.path.exists(pyenv):

    if script_name and file_to_execute:
        pyenv_exec_command = f'pyenv exec python "{file_to_execute}"'
        subprocess.run(pyenv_exec_command, shell=True)
    else:
        clear_screen()
        print_banner()

        try:
            versions = subprocess.check_output('pyenv versions', shell=True, text=True)
            print(f"\nPython Versions:\n {versions}")

            print(Fore.YELLOW + f"Usage: {script_name} <python_script.py>")
            input("\nPress Enter to Exit... ")

        except subprocess.CalledProcessError:
            pass

else:
    clear_screen()
    print_banner()
    print(Fore.RED + "\nPlease Install PyEnv-Win First!\n")
    input("\nPress Enter to Exit... ")

File: PyWrapper.pl

use strict;
use warnings;
use File::Copy;
use Win32::TieRegistry;
use Win32::Registry;
use Win32::Console::ANSI;
use Term::ANSIScreen qw/:color /;
use Term::ANSIScreen qw(cls);
use File::Basename;

my $clear_screen = cls(); 
my $version = "1.0.0";

my $BwE = (colored ['cyan on black'], qq{*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\n})
.(colored ['cyan on black'], qq{|}).(colored ['black on cyan'], qq{            __________          __________               }).(colored ['cyan on black'], qq{|\n})
.(colored ['cyan on black'], qq{|}).(colored ['black on cyan'], qq{            \\______   \\ __  _  _\\_   ____/               }).(colored ['cyan on black'], qq{|\n})
.(colored ['cyan on black'], qq{:}).(colored ['black on cyan'], qq{             |    |  _//  \\/ \\/  /|  __)_                }).(colored ['cyan on black'], qq{:\n})
.(colored ['cyan on black'], qq{.}).(colored ['black on cyan'], qq{             |    |   \\\\        //       \\               }).(colored ['cyan on black'], qq{.\n})
.(colored ['cyan on black'], qq{:}).(colored ['black on cyan'], qq{  (\\_/)      |______  / \\__/\\__//______  /               }).(colored ['cyan on black'], qq{:\n})
.(colored ['cyan on black'], qq{|}).(colored ['black on cyan'], qq{ ( ~_~)             \\/}).(colored ['black on cyan'], qq{PyEnv-Win Wrapper}).(colored ['black on cyan'], qq{\\/$version           }).(colored ['cyan on black'], qq{|\n})
.(colored ['cyan on black'], qq{|}).(colored ['black on cyan'], qq{ (>  >)                                                  }).(colored ['cyan on black'], qq{|\n})
.(colored ['cyan on black'], qq{*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\n});



my $script_name = $0;
my $file_to_execute = join(" ", @ARGV);

# Get the Windows drive (e.g., C:)
my $windows_drive = $ENV{'SystemDrive'} || 'C:';

# Get the username
my $username = $ENV{'UserName'} || '';

# Installed?
my $pyenv = '$windows_drive\\Users\\$username\\.pyenv';

# Check if this is the first run
if (!-e $pyenv) {

    if ($script_name && $file_to_execute) {
        my $pyenv_exec_command = 'pyenv exec python "' . $file_to_execute . '" ';
        system($pyenv_exec_command);
    } else {
		print $clear_screen;
		print $BwE;

		my $versions = qx(pyenv versions);
		print "\nPython Versions:\n $versions";

        print (colored ['bold yellow'], "\nUsage: $script_name <python_script.py>\n");
		print "\nPress Enter to Exit... ";
		
		while (<>) {
		chomp;
		last unless length;
		}
    }
	

} else {
	print $clear_screen;
	print $BwE;
	print (colored ['bold red'], "\nPlease Install PyEnv-Win First!\n");
	print "\nPress Enter to Exit... ";
	
	while (<>) {
	chomp;
	last unless length;
	}

}
